presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java

```diff
@@ -85,14 +85,12 @@ public DictionaryBlock(int positionCount, Block dictionary, Slice ids, boolean d
         else {
             int sizeInBytes = 0;
             int uniqueIds = 0;
-            boolean[] isReferenced = getReferencedPositions(dictionary, ids, positionCount);
-            for (int position = 0; position < isReferenced.length; position++) {
-                if (isReferenced[position]) {
-                    if (!dictionary.isNull(position)) {
-                        sizeInBytes += dictionary.getLength(position);
-                    }
-                    uniqueIds++;
+            for (int i = 0; i < positionCount; i++) {
+                int position = getIndex(ids, i);
+                if (!dictionary.isNull(position)) {
+                    sizeInBytes += dictionary.getLength(position);
                 }
+                uniqueIds++;
```

Comment:
Yep, I realized that's why the tests are failing. I'll update it and assign to Nileema